src/components/LayerSelector.vue

```diff
@@ -16,57 +16,145 @@ const emit = defineEmits<{
   (e: 'update:modelValue', value: string[]): void
 }>()
 
-const selections = ref<SelectableSingleItem[] | SelectableSingleItem[][]>([])
+const themeIdx = ref<number>()
+const themeItems = computed<any[]>(() =>
+  props.items?.filter((item: any) => item.id === 'theme')
+    .flatMap((item: any) => item.children)
+)
+
+const vulnerability = ref<string>()
+const vulnerabilityItems = computed<any[]>(() =>
+  props.items?.filter((item: any) => item.id === 'vulnerability')
+    .flatMap((item: any) => item.children)
+)
+
+const tab = ref<string>()
+const tabItems = computed<any[]>(() =>
+  props.items?.filter((item: any) => item.tab)
+)
+
+const timeIdx = ref<number>(0)
+const timeItems = [20, 50, 100]
+
+const scenarioIdx = ref<number>(0)
+const scenarioItems = [
+  { 
+    id: "base",
+    label: "Base"
+  }, 
+  { 
+    id: "cc",
+    label: "Climate change"
+  }]
+
+const showInundation = ref<boolean>(false)
 
-watch(selections.value, (value: SelectableSingleItem[]) => {
-  const newval = value
-    .filter((k: SelectableSingleItem) => k)
-    .map((k: SelectableSingleItem) => {
-      if (Array.isArray(k)) {
-        return k.map((val: SelectableSingleItem) => val.id)
-      } else {
-        return k.id
-      }
-    }).flat()
-  emit('update:modelValue', newval)
+watch([themeIdx, tab, showInundation, timeIdx, scenarioIdx, vulnerability], () => {
+  updateLayers()
 })
 
 watch(() => props.items,
   (value: SelectableItem[]) => {
-    value.forEach((item: SelectableItem, index: number) => {
-      if ('children' in item) {
-        const selectedChildren: SelectableSingleItem[] = item.children
-          .filter((child: SelectableSingleItem) => child.selected)
-        
-        selections.value[index] = item.multiple ? selectedChildren : selectedChildren.pop()
-      } else if (item.selected) {
-        selections.value[index] = item
-      }
-    })
+    // init with light theme
+    if (value.filter((item: SelectableItem) => item.id === 'theme').pop())
+      themeIdx.value = 1
```

Comment:
Shouldn't we put the default value in initialisation of ref ?

Reply:
using the selected property instead of hardcoding the "light" one